inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -85,205 +108,197 @@ cldnn::device_info clDNNEngine::GetDeviceInfo(const std::map<std::string, std::s
-            }
-            if (auto redOp = std::dynamic_pointer_cast<const ::ngraph::opset1::ReduceSum>(node)) {
-                auto reduction_axes = redOp->get_reduction_axes().to_vector();
-                bool reduce_along_f = redOp->get_reduction_axes().size() == 1 && std::count(reduction_axes.begin(), reduction_axes.end(), 1) != 0;
-                bool fp16_batch_not_1 = redOp->get_element_type() == ngraph::element::f16 && redOp->input(0).get_shape()[0] != 1;
-                bool can_use_reduce = !reduce_along_f && !fp16_batch_not_1;
-                return can_use_reduce;
-            }
+template<typename T>
+static bool disableReduceDecomposition(const std::shared_ptr<const ngraph::Node> node) {
+    if (auto op = std::dynamic_pointer_cast<const T>(node)) {
+        auto reduction_axes = op->get_reduction_axes().to_vector();
+        bool reduce_along_f = op->get_reduction_axes().size() == 1 && std::count(reduction_axes.begin(), reduction_axes.end(), 1) != 0;
+        bool fp16_batch_not_1 = op->get_element_type() == ngraph::element::f16 && op->input(0).get_shape()[0] != 1;
+        bool can_use_reduce = !reduce_along_f && !fp16_batch_not_1;
+        return can_use_reduce;
+    }
+    return false;
+}
 
-            if (auto add_op = std::dynamic_pointer_cast<const ngraph::opset1::Add>(node)) {
-                return ngraph::is_type<ngraph::opset1::Convolution>(add_op->get_input_node_shared_ptr(0)) ||
-                       ngraph::is_type<ngraph::opset1::GroupConvolution>(add_op->get_input_node_shared_ptr(0)) ||
-                       ngraph::is_type<ngraph::opset1::MatMul>(add_op->get_input_node_shared_ptr(0));
-            }
+InferenceEngine::CNNNetwork clDNNEngine::CloneAndTransformNetwork(const InferenceEngine::CNNNetwork& network,
+                                                                  const CLDNNPlugin::Config& config) const {
+    CNNNetwork clonedNetwork = InferenceEngine::details::cloneNetwork(network);
 
-            return std::dynamic_pointer_cast<const ::ngraph::opset2::Gelu>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset3::ShuffleChannels>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset2::BatchToSpace>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset2::SpaceToBatch>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset5::HSigmoid>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset4::HSwish>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset4::ReduceL1>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset4::ReduceL2>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset4::SoftPlus>(node) ||
-                   std::dynamic_pointer_cast<const ::ngraph::opset5::LogSoftmax>(node);
-        };
-        auto nGraphFunc = clonedNetwork->getFunction();
+    if (clonedNetwork.getFunction()) {
```

Comment:
Should we throw an exception or apply `ConstTransformer` if plugin get the legacy network?

Reply:
`cloneNetwork` above throw this exception. It works only with ngraph